Qualitative/BioGrid-based predictions of regulatory relationships

NElnour opened the pull request:
These tools are for BioGrid-based predictions of regulatory relationships using an interaction interpretation map developed on both GGI and PPI.

Note: someone used my fork to build their tool, so there are R functions here towards that end.

Commit:
added scripts for integrating BioGrid GGI and STRING PPI data of system into directed networks

Commit:
annotated scripts with authorships

Commit:
edited readme with specifications

Commit:
added option for user to display stringent vs relaxed interaction network

Commit:
visualize relaxed networks

Commit:
embedded network view

Commit:
fixed readme

Commit:
relaxed network view

Commit:
added network view

Commit:
updated README for ease of integration

Commit:
Update README.md

Commit:
fixed biogridr credentials

Commit:
wrote getKey to set up user credentials

Commit:
Merge branch 'master' of https://github.com/NElnour/BCB420.2019.ESA

Commit:
added GGI interpretation map

Commit:
attempt to resolve Travis build failure

Commit:
replaced with original files

Commit:
resolved warnings

Commit:
added import statements to Description

Commit:
added roxygen2-formatted documentation

Commit:
resolved roxygen2 dependencies

Commit:
removed old .Rd files

Commit:
removed examples from .Rd files

Commit:
renamed tool

Commit:
removed old .rd files

Commit:
Resolved roxygen2 errors and warnings

Commit:
updated documentation with import statements

Commit:
added import support for remote package biogridr

Commit:
resolved import clashes

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
network hypothesis

Commit:
Update README.md

Commit:
refactored code to remove dependency on external datasets; generated updated .Rd files

Commit:
fixed ratification problem

Commit:
reverted biogridr changes

Commit:
fixed example building issues

Commit:
updated readme

Commit:
multisystem example added

Commit:
updated readme annotations

Commit:
fixed readme example

Commit:
reolved merge conflicts

Merge branch 'master' of git://github.com/hyginn/BCB420.2019.ESA

# Conflicts:
#	NAMESPACE
#	R/SyDButils.R
#	R/fetchComponents.R
#	R/fetchData.R
#	README.md

Commit:
removed merge markers

Commit:
refactored exported functions into separate files

Commit:
changed formatting per reviewer's suggestions

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	README.md

Commit:
removed unecessary imports and refactored getSysInteractions to remove biogridr dependency

Commit:
added comments to functions

Commit:
removed collation

Commit:
updated documentation and simplified hypothesize.R

Commit:
I think someone installed my fork and made changes. I resolved the merge conflict.

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	DESCRIPTION
#	NAMESPACE

Commit:
fixed annotation maps

Commit:
standardized column names

Commit:
updated readme

Commit:
fixed readme and examples

sapirl commented on DESCRIPTION:
```diff
@@ -24,6 +28,10 @@ LazyData: true
 Imports:
     Biostrings,
+    utils,
+    stats,
```
No need to add base R packages

NElnour replied:
fixed

sapirl commented on R/getGeneticInteractome.R:
```diff
@@ -0,0 +1,42 @@
+#' \code{getGeneticInteractome} augments systems with BioGRID genetic
+#' interaction annotations
+#'
+#' @param mySys The dataframe of PPI between system's components
```
The @param and @return should match the format described here
https://github.com/hyginn/BCB420.2019.ESA/blob/master/PullRequestReviewNotes.md

NElnour replied:
fixed

sapirl commented on R/getGeneticInteractome.R:
```diff
@@ -0,0 +1,42 @@
+#' @return The dataframe of system components and either their GGI between
+#' physical interactors should be selected (iff \code{criterion} == "stringent")
+#' or all GGI of physical interactors (iff \code{relaxed} == "stringent")
+#'
```
@example + short explanation is needed for every exported function

NElnour replied:
fixed

sapirl commented on R/hypothesize.R:
```diff
@@ -0,0 +1,93 @@
+#' components of \code{mySys} and edges are GGI interpretations in either
+#' \code{EMAP} or \code{GMAP}
+#'
+#' @param network A dataframe of physically-interacting system components with
```
As above: changing the parameters and results according to this format
https://github.com/hyginn/BCB420.2019.ESA/blob/master/PullRequestReviewNotes.md

NElnour replied:
fixed

sapirl commented on R/hypothesize.R:
```diff
@@ -0,0 +1,93 @@
+#' @param gmap The dataframe obtained from \code{makeGMAP}.
+#' @param emap The dataframe obtained from \code{makeEMAP}.
+#'
+visualizeInteractions <- function(network, emap, ppi_ggi, gmap) {
```
The return value needs to be explicit, by using return()

NElnour replied:
fixed...although the function itself does not return anythin; it just plots and formats the graphs'  options

sapirl commented on R/hypothesize.R:
```diff
@@ -0,0 +1,93 @@
+#'
+#' @inheritParams hypothesize
+#' @param gmap The dataframe obtained from \code{makeGMAP}.
+#' @param emap The dataframe obtained from \code{makeEMAP}.
```
The @return is missing.

Also, as above, changing the parameters format

NElnour replied:
fixed.

sapirl replied:
Great! after fixing every thing, please submit a new pull request (that way the tag for "changes requested" will be removed)

sapirl commented on R/makeMaps.R:
```diff
@@ -0,0 +1,110 @@
+
+#' Generate the genetic interpretation map of BioGrid GGI tags of system's physical interactions.
+#'
+#' @return The dataframe mapping BioGrid GGI tag to its interpretation assuming that the system's components also interact physically.
```
Needs to match the format
https://github.com/hyginn/BCB420.2019.ESA/blob/master/PullRequestReviewNotes.md

NElnour replied:
Fixed.

sapirl commented on R/makeMaps.R:
```diff
@@ -0,0 +1,110 @@
+
+#' Generate the genetic interpretation map of BioGrid GGI tags.
+#'
+#' @return The dataframe mapping BioGrid GGI tag to its interpretation.
```
As above

sapirl commented on R/predictDirectedNetworks.R:
```diff
@@ -0,0 +1,63 @@
+# predictDirectedNetworks.R
+
+source('./R/fetchData.R', echo=TRUE)
```
No need to source those files, as they are installed and loaded whenever as part of the package

NElnour replied:
Fixed.

sapirl commented on R/predictDirectedNetworks.R:
```diff
@@ -0,0 +1,63 @@
+#' \code{getSysInteractions} filter system by physically-interacting components
+#' and return their genetic interactions
+#'
+#' @param sysName A string specifying the full path to the excel sheet
```
As above: format

NElnour replied:
Fixed.

sapirl commented on R/predictDirectedNetworks.R:
```diff
@@ -0,0 +1,63 @@
+#' @import utils
+#' @include SyDButils.R fetchData.R
+#' @importFrom stats complete.cases
+#' @examples
```
A short explanation is required for each example of an exported function

NElnour replied:
Fixed.

sapirl reviewed (changes requested):
The comments

sapirl reviewed (changes requested):
The changes

Commit:
added tool vignette

Commit:
resolved reviewer's changes

Commit:
resolved merge issues

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	NAMESPACE
#	R/jaccard_dist.R
#	man/jaccard_dist.Rd

Commit:
reformatted documentation as required

Commit:
removed comma from description author

Commit:
fixed vignette

Commit:
resolved merge conflicts

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	DESCRIPTION
#	NAMESPACE

Commit:
fixed author under description

Commit:
moved image files to extdata/img and reduced their sizes
